src/rpc/blockchain.cpp

```diff
@@ -665,7 +665,9 @@ UniValue getblockchaininfo(const UniValue& params, bool fHelp)
     if (fPruneMode)
     {
         CBlockIndex *block = chainActive.Tip();
-        while (block && block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA))
+        assert(block);
+        
+        while (block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA))
             block = block->pprev;
 
         obj.push_back(Pair("pruneheight",        block->nHeight));
```

Comment:
@luke-jr I think it won't always be `0` since `& BLOCK_HAVE_DATA` would fail for the first pruned block, hence,  the last "pruned" block.